### PR TITLE
Update Unliftable API

### PR DESCRIPTION
**Pull request by nicolasstucki**
* Align with Liftable API definitions
* Simplify implementation signatures
* Add docs
* Align Liftable/Unliftable names

### CODE REVIEW

File: library/src-bootstrapped/scala/quoted/Unliftable.scala [hunk 1]
```diff
@@ -8,368 +8,365 @@ trait Unliftable[T] {
    *  Returns `None` if the expression does not contain a value or contains side effects.
    *  Otherwise returns the `Some` of the value.
    */
-  def apply(x: Expr[T])(using qctx: QuoteContext): Option[T]
+  def fromExpr(x: Expr[T]): QuoteContext ?=> Option[T]
```
Comment (liufengyun) [inline library/src-bootstrapped/scala/quoted/Unliftable.scala:11]: What about the name `from`? `unlift.from(expr)` reads better than `unlift.fromExpr(expr)`.

```suggestion
  def from(x: Expr[T]): QuoteContext ?=> Option[T]
```

**Review by liufengyun — APPROVED** [review]
LGTM

File: library/src-bootstrapped/scala/quoted/Unliftable.scala (hunk 1 above)
Comment (nicolasstucki) [reply library/src-bootstrapped/scala/quoted/Unliftable.scala:11]: That would not align with the `Liftable` and we almost never need to write it explicitly.

Comment (liufengyun) [reply library/src-bootstrapped/scala/quoted/Unliftable.scala:11]: > That would not align with the `Liftable` and we almost never need to write it explicitly.

It makes sense to be consistent.

**Pull request merged**